docs/design/2023-12-23-index-usage.md

```diff
@@ -0,0 +1,185 @@
+
+The percentage is calculated by dividing the count of accessed rows by the count of all rows in the table. The count of rows in table is an assumption value by managing table row deltas, which has already been used by optimizer.
+
+These data are all stored in TiDB memory. Therefore, the data on an instance are lost when the TiDB instance shutdown or reboot.
+
+Also, we provide a view to help users identify the unused index quickly: the `sys.schema_unused_indexes`. It contains the following columns:
+
+|Field|DataType|Description|
+|-|-|-|
+|TABLE_SCHEMA|varchar(65)|The corresponding `schema` of the table|
+|TABLE_NAME|varchar(65)|The name of the table|
+|INDEX_NAME|varchar(65)|The name of the index|
+
+It relies on the `information_schema.cluster_tidb_index_usage`. It'll list the indexes which haven't been used since the last startup of tidb instances.
+
+### Data source
+
+We need to get / calculate the following data, and record them in a per-node memory structure:
+
+1. `KV_REQ_TOTAL`. It is equal to the `totalTasks` in the `basicCopRuntimeStats`. A new method is needed to expose it.
+2. `QUERY_TOTAL`. We need a structure on `StmtCtx` to make sure a query is only counted once.
+3. `READ_SELECTION_RATE`, `ACCESS_SELECTION_RATE`. They are stored in a histogram, and are calculated by access_rows / total_rows.
+
+  1. The rows returned by the following plan should be recorded as `access_rows`
+     1. `PhysicalIndexScan`
+     2. `PointGet` if the object is an index.
+  2. `total_rows` can be dumped from `(*statistics.HistColl).RealtimeCount`.The RealtimeCount in `sc.usedStatsInforecords` the used stats in current statement.
+
+The existing `RuntimeStatsColl` records the execution summary of each physical plan, and is accessed by physical plan ID. We can have multiple options on when / how to re-order the data into index usages.
+
+These physical plans have a corresponsing executor. We need to consider the following executors:
+
+1. `IndexReaderExecutor`
+2. `IndexLookUpExecutor`
+3. `IndexMergeReaderExecutor`
+4. `PointGetExecutor`
+5. `BatchPointGetExecutor`
+
+Then the rows returned by the `Index(Range|Full)Scan` inside can be recorded as `access_rows`.
+
+This information is reported and collected when the corresponding executor is closed. For example:
+
+```go
+func (e *IndexLookUpExecutor) Close() error {
+    // ...
+    accessRows := runtimeStatsColl.GetCopStats(e.idxPlans[0].ID()).GetActRows()
+    returnRows := accessRows
+    if len(e.idxPlans) > 1 {
+        selID := e.idxPlans[len(e.idxPlans)-1].ID()
+        returnRows = runtimeStatsColl.GetCopStats(selID).GetActRows()
+    }
+    // record the `accessRows` and `returnRows`
+    // ...
+```
+
+### Index Usage Collector
+
+We already have a similar RFC [2020-09-30-index-usage-information.md](https://github.com/pingcap/tidb/blob/master/docs/design/2020-09-30-index-usage-information.md). It gives a good idea of collecting index usages without locking.
```

Comment:
Well, since we have [2020-09-30-index-usage-information.md](https://github.com/pingcap/tidb/blob/master/docs/design/2020-09-30-index-usage-information.md) already, how about update it by adding a 'Update' section for all of this RFC? It's weird that we have 2 individual RFC docs for 'index uage'.